src/base-widget.ts

```diff
@@ -3,27 +3,44 @@ import { isPlatformBrowser } from '@angular/common';
 import { bem, noop } from './utils';
 import { NgAisInstantSearch } from './instantsearch/instantsearch';
 import { NgAisIndex } from './index-widget/index-widget';
-import { Widget } from 'instantsearch.js/es/types';
-export { Widget };
-
-// TODO: use Connector type from InstantSearch. Not yet possible now,
-// since non-ts connectors can't have generics like Connector has,
-// as well as sometimes being not accurate enough / missing keys.
-export type Connector = (
-  renderFn: (state: object, isFirstRendering: boolean) => void,
-  unmountFn: () => void
-) => (widgetOptions?: object) => Widget;
-
-export abstract class BaseWidget implements OnInit, OnDestroy {
+import {
+  Widget,
+  WidgetDescription,
+  Connector,
+} from 'instantsearch.js/es/types';
+
+export { Widget, Connector };
+
+export type ExtractConnectorParams<
+  T extends Connector<unknown & WidgetDescription, unknown>
```

Comment:
could we create these out of the widget's WidgetDescription instead of Connector? It should be easier and less need for inferring things, wdyt?

Reply:
It's more parameters and requires knowing some of the typing internals but it's much much more simple, hence better. Tried that here : a6047e7

Reply:
that commit looks much nicer, perfect!